File: Java/src/Programmers/Test/primenum.java
package Programmers.Test;

public class primenum {
    public int solution(int n) {
        int answer = 0;
        System.out.println(n);
        boolean result = false;
        for(int i = 2; i <= n; i++){
            result = isPrime(i);
            System.out.println(i + " " + result);
            if(result){
                answer++;
            }
        }

        return answer;
    }

    private boolean isPrime(int num){
        boolean result = true;
        if((num % 2 == 0 && num != 2) | num == 1) {
//            System.out.println(num + " " + result);
            result = false;
        } else {
            for (int i = 2; i * i <= num; i++) {
                if(num % i != 0) {
                    result = true;
//                    System.out.println(i + " " + result);
                } else {
                    result = false;
//                    System.out.println(i + " " + result);
                    break;
                }
            }
        }
        return result;
    }

    public static void main(String[] args) {
        int numbers = 5;
        primenum foundPrimeNum = new primenum();
        System.out.println(foundPrimeNum.solution(numbers));
    }
}

File: Python/Programmers/Level 1/prime_num.py
def solution(n):
    answer = 0

    for i in range(1, n+1):
        answer += 1 if is_prime(i) else 0

    return answer


def is_prime(n):
    if n == 1 or (n != 2 and n % 2 == 0):
        return False
    elif n == 2:
        return True
    e = round(n ** 0.5) + 1
    for i in range(3, e, 2):
        if n % i == 0:
            return False
    return True


print(solution(10))
